build/premake5.lua

```diff
@@ -11,7 +11,7 @@ project "rive"
 
     files {"../src/**.cpp"}
 
-    buildoptions {"-Wall", "-fno-exceptions", "-fno-rtti"}
+    buildoptions {"-Wall", "-fno-exceptions", "-fno-rtti", "-Werror=format"}
```

Comment:
I'd say using this flag is a good win for applications using printf formatting.

The issue manifests itself in two unwanted forms:
1) The wrong output (confusing when debugging)
2) The application crashes at runtime